contrib/kube-prometheus/hack/cluster-monitoring/deploy

```diff
@@ -32,6 +32,12 @@ kctl apply -f manifests/prometheus/prometheus-k8s-service.yaml
 kctl apply -f manifests/alertmanager/alertmanager-config.yaml
 kctl apply -f manifests/alertmanager/alertmanager-service.yaml
 
+# unfortunately statefulsets cannot be changed except for their replica count
+# so we need to make sure that the rule files are created before we create the
+# prometheus resource so it can properly discover the rule files when creating
+# the statefulset
+sleep 5
```

Comment:
Actually a sidecar won't be enough, as what happens is that the `StatefulSet` gets created without the rule file `ConfigMap`s mounted and then the Prometheus Operator wants to modify the `PodTemplate`, which is an invalid action on `StatefulSet`s (only replicas and containers may be changed).

What we could do: if `ruleSelector` is specified, we could not create the `StatefulSet` at all until `ConfigMap`s selected by the `ruleSelector` are found. That would not require any new RBAC requirements. If we agree on this I'm happy to add it to this PR.